guardian/guardian.go

```diff
@@ -0,0 +1,196 @@
+package guardian
+
+import (
+	"context"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/cipherowl-ai/addressdb/address"
+	"github.com/cipherowl-ai/addressdb/reload"
+	"github.com/cipherowl-ai/addressdb/securedata"
+	"github.com/cipherowl-ai/addressdb/store"
+)
+
+const (
+	bloomFilterFilename = "bloom_filter.gob"
+	privateKeyFilename  = "privkey.asc"
+	publicKeyFilename   = "pubkey.asc"
+
+	linuxPath  = ".story/geth/guardian"
+	darwinPath = "Library/Story/geth/guardian"
+)
+
+var (
+	instance  *Guardian
+	initOnce  sync.Once
+	basicPath = getDefaultPath()
+
+	// Path to the private key file
+	privateKeyPath = filepath.Join(basicPath, privateKeyFilename)
+	// Path to the public key file
+	publicKeyPath = filepath.Join(basicPath, publicKeyFilename)
+	// Passphrase for the private key
+	passphrase = "CpRBS9GTL2tR0ZO6"
+)
+
+// Guardian provides transaction filtering to prevent interactions with certain pre-defined addresses.
+type Guardian struct {
+	filter  *store.BloomFilterStore // Stores addresses that are filtered
+	manager *reload.ReloadManager   // Manages reloading of the filter data
+	mu      sync.Mutex              // Mutex to control access to the Guardian's operations
+}
+
+// Config represents the configuration for initializing Guardian.
+type Config struct {
+	Enabled        bool   // If true, the Guardian will filter transactions
+	FilterFilePath string // File path to the bloom filter file
+}
+
+// DefaultConfig contains the default configuration for Guardian.
+var DefaultConfig = Config{
+	FilterFilePath: filepath.Join(basicPath, bloomFilterFilename),
+}
+
+// InitInstance initializes a singleton instance of the Guardian with the given configuration.
+// If the configuration disables the Guardian, the instance is not created.
+func InitInstance(config Config) {
+	initOnce.Do(func() {
+		if !config.Enabled {
+			log.Warn("Guardian is disabled")
+			return
+		}
+
+		var err error
+		instance, err = newGuardian(config)
+		if err != nil {
+			log.Error("Failed to initialize Guardian", "error", err)
+			return
+		}
+	})
+}
+
+// GetInstance returns the singleton Guardian instance.
+func GetInstance() *Guardian {
+	return instance
+}
+
+// newGuardian creates a new Guardian instance from the provided config.
+func newGuardian(config Config) (*Guardian, error) {
+	// Create the secure data handler
+	handler, err := securedata.NewPGPSecureHandler(
```

Comment:
we prefer not using `secureHandler` currently. 
let's delete all related code and test without `secureHandler`

Reply:
Done